runtime/src/main/java/org/corfudb/runtime/collections/CorfuQueue.java

```diff
@@ -149,10 +149,27 @@ public String toString() {
      *          known until the transaction commits.
      *          The ID returned here is only really useful for remove() operations.
      */
-    public CorfuRecordId enqueue(E e) {
+    public void enqueue(E e) {
         final Long id = guidGenerator.nextLong();
-        corfuTable.put(id, e);
-        return new CorfuRecordId(0, id);
+        CorfuQueueRecord<E> queueEntry = new CorfuQueueRecord<>(0, 0, id, e);
```

Comment:
Is it better to use invalid epoch and sequence instead of 0 ?

Reply:
would it matter? keeping it as zero allows these non-transactional entries to go to the first of the queue - might make it easier for testing

Reply:
But then if you compare the a non-tranactional entry with a tranactional entry who was committed at epoch 0, sequence 0,  it would be confusing right?

Reply:
yes, that isn't supported. ok let me catch that case and assert.